### ISSUE TITLE
Is this project still active?

### ISSUE THREAD

**Issue by sagiegurari** [carlos8f/haredis#27]
I really need a HA solution for redis and it seemed like the most suitable project out there, but it seems its not being developed for some time, the build is failing in the tests on some timeout and it's using a very old node_redis version (its up to 0.12.1 already).

Are there any plans to continue development on this project?
if not, any good alternatives?

**Comment by carlos8f** [comment]
Hello,

Unfortunately you’re right about the project being out of date. However as far as I know, the project still works as intended and the tests failing is not a good indication of that. The tests were ported over from node_redis, and at the time they were ported, they were failing in some environments due to race conditions and/or dependent on a particular redis-server version. I attempted to fix them after porting, but porting the tests **and** fixing them every time they changed in the node_redis repository was not very practical. I tried updating to new node_redis versions, but the API changed and broke the wrapper code and tests, which I didn’t have time to rewrite. Also travis-ci had issues running multiple redis servers and ports.

Although I have used haredis in production (and still do) without incident, there is also the redis cluster/sentinel project that is similar but doesn’t accomplish all the same things (namely realtime notification and switching of the node app’s connections to the new master). I haven’t been following redis cluster development very closely since haredis has served my availability/load balancing purposes fine and I haven’t had the bandwidth or need to keep up with subsequent node_redis releases.

I’d invite you to try haredis if you find these caveats acceptable. If you find something better, I’d also like to hear from your experience.

Cheers,
Carlos
